### PR TITLE
Remove duplication of ssh keys and fix usage on PR builds

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Cherry picked from #4947.

Before, each PR contains the following block at the end of the logs:
```
0.04s$ bash util/travis_docs.sh
iv undefined
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
chmod: cannot access ‘id_rsa_qmk_firmware’: No such file or directory
chmod: cannot access ‘id_rsa_qmk.fm’: No such file or directory
Agent pid 10110
id_rsa_qmk_firmware: No such file or directory
The command "bash util/travis_docs.sh" exited with 0
```

While fixing the above, i also reduced the duplication of ssh keys handling within travis_compiled_push.sh

## Types of changes

- [ ] Core
- [x] Bugfix
- [ ] New Feature
- [ ] Enhancement/Optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/Layout/Userspace (addition or update)
- [ ] Documentation

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document. (https://docs.qmk.fm/#/contributing)
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: util/travis_push.sh [hunk 1]
```diff
@@ -5,13 +5,15 @@ TRAVIS_PULL_REQUEST="${TRAVIS_PULL_REQUEST:false}"
 TRAVIS_COMMIT_MESSAGE="${TRAVIS_COMMIT_MESSAGE:-none}"
 TRAVIS_COMMIT_RANGE="${TRAVIS_COMMIT_RANGE:-HEAD~1..HEAD}"
 
-git config --global user.name "QMK Bot"
-git config --global user.email "hello@qmk.fm"
+if [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_PULL_REQUEST" == "false" ]] ; then
+	git config --global user.name "QMK Bot"
```
Comment (mechmerlin) [inline util/travis_push.sh]: Is setting it in `--global` necessary?

Comment (zvecr) [reply util/travis_push.sh]: @mechmerlin not sure i makes much difference on travis, as its a clean vm per build. I kept as much of the logic the same, to minimise the change as much as i could.

**Comment by drashna** [comment]
@jackhumbert or @skullydazed ?

**Comment by drashna** [comment]
Ugh, I think this broke travis, partially.

In the past, we've had issues trying to move this block:
```bash
TRAVIS_BRANCH="${TRAVIS_BRANCH:master}"	source util/travis_push.sh
TRAVIS_PULL_REQUEST="${TRAVIS_PULL_REQUEST:false}"	
TRAVIS_COMMIT_MESSAGE="${TRAVIS_COMMIT_MESSAGE:-none}"	
TRAVIS_COMMIT_RANGE="${TRAVIS_COMMIT_RANGE:-HEAD~1..HEAD}"	
```

And it looks like it's no longer detecting the type of changes (keyboard, keymap, other)

**Comment by zvecr** [comment]
@drashna if you can revert the merge so the behaviour is not broken for everyone, i will look into why this is failing.

**Comment by zvecr** [comment]
@drashna just to confirm, the issue you are detecting is with the `travis_build.sh` build selection logic? Because that file _should be_ unaffected by this change, either directly by modifications, or behaviour ( as `travis_build.sh` does not source or call `travis_compiled_push.sh` or `travis_push.sh`).

**Comment by drashna** [comment]
To be honest, this is a guess, based on the last time that it broke.
Basically, I think it's not picking up the variables correctly. 

And I think that all we need is something like this:
https://github.com/qmk/qmk_firmware/compare/master...drashna:fix/travis

Not a full revert.

**Comment by zvecr** [comment]
Not sure those changes will fix anything, given it wont touch the `travis_build.sh` script.

#5127 was before my changes were merged, and this also suffers from not building the keymap.
The changes in that pull are only keymap and this is the build output
```bash
0.02s$ bash util/travis_build.sh
keyboards/kc60/keymaps/noroadsleft/keymap.c
keyboards/kc60/keymaps/noroadsleft/readme.md
keyboards/kc60/keymaps/noroadsleft/readme_ch1.md
keyboards/kc60/keymaps/noroadsleft/readme_ch2.md
keyboards/kc60/keymaps/noroadsleft/readme_ch3.md
keyboards/kc60/keymaps/noroadsleft/readme_ch4.md
keyboards/kc60/keymaps/noroadsleft/readme_ch5.md
keyboards/kc60/keymaps/noroadsleft/readme_git.md
The command "bash util/travis_build.sh" exited with 0.
```

**Comment by drashna** [comment]
Okay, then I may be wrong about the timeline.  It was an guess based on past experience. 
Eg, this isn't the first time that we've broken travis by moving out the defines here: https://github.com/qmk/qmk_firmware/pull/2642

However, I'm not 100% sure of the cause, but the `travis_compiled_push.sh` script is what does determine *what* to compile against., IIRC.

**Comment by zvecr** [comment]
Its definitely `travis_build.sh` that contains the logic, i hit issues with it during my docker work. I will have a look into it. Given it was already broken, can we hold off on any partial/full reverts.

**Comment by drashna** [comment]
Not a problem.  If it can be fixed, that's awesome.

**Comment by zvecr** [comment]
Raised #5160 for the issue observed.

**Pull request merged**